pkg/util/provider/machinecodes/status/status.go

```diff
@@ -88,10 +107,18 @@ func FromError(err error) (s *Status, ok bool) {
 
 	if matches, errInFind := findInString(err.Error()); errInFind == nil {
 		code := codes.StringToCode(matches[0])
-		return New(code, matches[1]), true
+		return &Status{
+			code:    int32(code),
+			message: matches[1],
+			cause:   err,
```

Comment:
here the `cause` shouldn't be filled with `err` as its the `status.Status` object only. in case we can't decode the `err` then it should be filled, like you have done when the `if condition` fails.
I think we should keep it nil and explain with a comment abt the reason, WDYT?

Reply:
Ok that is right.